server/validators/goalReviewDateValidator.ts

```diff
@@ -0,0 +1,15 @@
+import moment from 'moment'
+
+export default function validateGoalReviewDate(reviewDate?: Date): Array<string> {
+  const errors: Array<string> = []
+
+  if (!reviewDate) {
+    errors.push('Enter a review date for the goal')
+  } else if (Number.isNaN(reviewDate.getTime())) {
+    errors.push('Enter a review date in the correct format')
+  } else if (reviewDate < moment().toDate()) {
+    errors.push('Enter a review date in the future')
```

Comment:
Are you thinking of making this more generic for when we validate a step's target date?

Reply:
Not quite. I've made it generic (more specifically agnostic) to the form where a goal review date will need to be validated (ie. create goal, and later on, edit goal)

I'm envisaging we will have a very similar validator function, but for step target date, and it will have different error message wording. (or, perhaps we wont have it at all, and it will be a "step target date range validator" 🙄 )

Reply:
I was thinking it would be nice if the message came from somewhere else. The validator would know how to get the message based on the field name. For example, I'm thinking of what we used to do with `ResourceBundles`. It's been a long time since I used one of those! 😄